pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/plan/PhysicalPlanVisitor.java

```diff
@@ -96,10 +102,85 @@ public MultiStageOperator visitMailboxSend(MailboxSendNode node, PhysicalPlanCon
   @Override
   public MultiStageOperator visitAggregate(AggregateNode node, PhysicalPlanContext context) {
     MultiStageOperator nextOperator = node.getInputs().get(0).visit(this, context);
+    DataSchema inputSchema = node.getInputs().get(0).getDataSchema();
+    DataSchema resultSchema = node.getDataSchema();
+
+    // Convert aggCalls to FunctionContext and ExpressionContext that our aggregation functions understand.
```

Comment:
Move this logic to convert RexExpression to FunctionContext/ExpressionContext -> NewAggregateOperator.java